File: backend/autotest/views/system/roles.py
from flask import Blueprint, request

from autotest.services.sys_services.role import RolesService
from autotest.utils.api import partner_success

bp = Blueprint('roles', __name__, url_prefix='/api/roles')


@bp.route('/list', methods=['POST'])
def all_roles():
    """
    获取所有角色数据
    :return:
    """
    data = RolesService.list(**request.json)
    return partner_success(data=data)


@bp.route('/saveOrUpdate', methods=['POST'])
def save_or_update():
    """
    新增或更新角色
    :return:
    """
    RolesService.save_or_update(**request.json)
    return partner_success()


@bp.route('/deleted', methods=['POST'])
def deleted():
    """
    删除角色
    :return:
    """
    parsed_data = request.json
    r_id = parsed_data.get('id', None)
    RolesService.deleted(r_id)
    return partner_success()

File: frontend/src/api/useSystemApi/roles.ts
import request from '/@/utils/request';

/**
 * 用户接口
 * @method getUserList 获取用户列表
 * @method allMenu 获取菜单接口，平铺
 * @method saveOrUpdateMenu 更新保存菜单
 */
export function useRolesApi() {
  return {
    getList: (data?: object) => {
      return request({
        url: '/roles/list',
        method: 'POST',
        data,
      });
    },
    saveOrUpdate(data?: object) {
      return request({
        url: '/roles/saveOrUpdate',
        method: 'POST',
        data
      })
    },
    deleted(data?: object) {
      return request({
        url: '/roles/deleted',
        method: 'POST',
        data
      })
    }
  };
}
